Source/Caches/HACache.swift

```diff
@@ -381,7 +408,7 @@ public class HACache<ValueType> {
                 // No subscribers, do not connect.
                 return
             }
-            guard !state.isWaitingForPopulate else {
+            guard !state.isWaitingForPopulate || populateInfo == nil else {
```

Comment:
this feels like an invariant breakage that might be confusing later. suggest you do one of:

1. make startPopulate noop and do the normal code path for this case
2. set isWaitingForPopulate to false when starting the subscription for a populate-less subscription
3. make isWaitingForPopulate an enum with a case for not issuing populates, and don't change it below when in that case (e.g. `mutating func started()` changes the case around and checks)

Reply:
I think the way it is written now is correct, it reads like "If it is not waiting for populate or if there is no populate then move forward", I can add a comment if that helps.

Before doing that I tried number "2", but then I got those issues with reentrant access in cache